### PR TITLE
Added filtered_input and constrained_decoding

### LINKED ISSUE #607: Add constrained decoding to CRF
**Describe the feature and the current behavior/state.**
We can reuse code from the `crf_multitag_sequence_score` to allow constrained decoding. 

**Relevant information**
- Are you willing to contribute it (yes/no): yes
- Are you willing to maintain it going forward? (yes/no): yes
- Is there a relevant academic paper? (if so, where): we only need to reuse the functions in `tfa.text.crf` to get this functionality
- Is there already an implementation in another framework? (if so, where): yes easy to extend `tfa.text.crf`
- Was it part of tf.contrib? (if so, where): 

**Which API type would this fall under (layer, metric, optimizer, etc.)**
This will be under `tfa.text.crf`

**Who will benefit with this feature?**
People in NLP community, constrained decoding is important to deal with padding issues as well to denote start and end of a sequence. 

**Any other info.**

### PR DESCRIPTION
**Pull request by napsternxg**
# Description

Brief Description of the PR:

We can have a common function for creating filtered_inputs used in `crf_multitag_sequence_score`
This function can be reused to modify the input to `crf_decode` to support constrained decoding.

Fixes #607

## Type of change

- [ ] Bug fix
- [ ] New Tutorial
- [x] Updated or additional documentation
- [x] Additional Testing
- [ ] New Activation and the changes conform to the [activation contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/activations/README.md#contribution-guidelines)
- [ ] New Callback and the changes conform to the [callback contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/callbacks/README.md#contribution-guidelines)
- [ ] New Image addition and the changes conform to the [image op contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/image/README.md#contribution-guidelines)
- [ ] New Layer and the changes conform to the [layer contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/layers/README.md#contribution-guidelines)
- [ ] New Loss and the changes conform to the [loss contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/losses/README.md#contribution-guidelines)
- [ ] New Metric and the changes conform to the [metric contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/metrics/README.md#contribution-guidelines)
- [ ] New Optimizer and the changes conform to the [optimizer contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/optimizers/README.md#contribution-guidelines)
- [ ] New RNN Cell and the changes conform to the [rnn contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/rnn/README.md#contribution-guidelines)
- [ ] New Seq2seq addition and the changes conform to the [seq2seq contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/seq2seq/README.md#contribution-guidelines)
- [x] New Text addition and the changes conform to the [text op contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/text/README.md#contribution-guidelines)

# Checklist:

- [x] I've properly [formatted my code according to the guidelines](https://github.com/tensorflow/addons/blob/master/CONTRIBUTING.md#coding-style)
    - [x] By running Black + Flake8
    - [ ] By running pre-commit hooks
- [x] This PR addresses an already submitted issue for TensorFlow Addons
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] This PR contains modifications to C++ custom-ops

# How Has This Been Tested?

If you're adding a bugfix or new feature please describe the tests that you ran to verify your changes:
*  I ran pytest locally to test this PR. All tests passed.

### CODE REVIEW

**Comment by napsternxg** [comment]
@facaiy @Squadrick @seanpmorgan can you have a look at it?

File: tensorflow_addons/text/crf.py [hunk 1]
```diff
@@ -24,6 +24,29 @@
...
+    constrained to a selected set of tags.
+    Args:
+      inputs: A [batch_size, max_seq_len, num_tags] tensor of unary potentials
+          to use as input to the CRF layer.
+      tag_bitmap: A [batch_size, max_seq_len, num_tags] boolean tensor
+          representing all active tags at each index for which to calculate the
+          unnormalized score.
+    Returns:
+      filtered_inputs: A [batch_size] vector of unnormalized sequence scores.
+    """
```
Comment (WindQAQ) [inline tensorflow_addons/text/crf.py:40]:
```python
def crf_filtered_inputs(inputs: TensorLike, tag_bitmap: TensorLike) -> tf.Tensor:
     """Constrains the inputs to filter out certain tags at each time step.

     tag_bitmap limits the allowed tags at each input time step.
     This is useful when an observed output at a given time step needs to be
     constrained to a selected set of tags.

     Args:
       inputs: A [batch_size, max_seq_len, num_tags] tensor of unary potentials
           to use as input to the CRF layer.
       tag_bitmap: A [batch_size, max_seq_len, num_tags] boolean tensor
           representing all active tags at each index for which to calculate the
           unnormalized score.
     Returns:
       filtered_inputs: A [batch_size] vector of unnormalized sequence scores.
     """
```

**Review by WindQAQ — COMMENTED** [review]
LGTM! One minor documentation issue.

File: tensorflow_addons/text/crf.py (hunk 1 above)
Comment (napsternxg) [reply tensorflow_addons/text/crf.py]: Thanks @WindQAQ fixed it.

**Comment by WindQAQ** [comment]
@napsternxg Can you run the formatting tools again? I'd like to merge this after the tests pass. Thank you!

**Comment by napsternxg** [comment]
@WindQAQ Fixed it.

**Review by WindQAQ — APPROVED** [review]
Thank you!

**Comment by napsternxg** [comment]
@Squadrick was removed by mistake when I clicked on request review.

**Pull request merged**